src/components/hooks/useElementPosition.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * Custom hook that returns an element reference and the position of the
+ * element it refers to.
+ *
+ * @param {object} elementRef  Reference to the element
+ *
+ * @returns {DOMRect} Element position
+ */
+export default function useElementPosition(elementRef) {
+  const target = elementRef.current;
+  let elementPosition;
+
+  if (target) {
```

Comment:
this should be inside an `useEffect` so that we know that getBoundingClientRect() is called after the dom has been updated https://reactjs.org/docs/hooks-effect.html#effects-without-cleanup

```js
useEffect(() => {
  if (target) {
    setRect(target.getBounding());
  }
});

useEffect(() => {
   window.addEventListener();
   return () => window.removeEventListener();
}, []);
```

the first effect executes after each render, the second one just once when the component mounts.